### PR TITLE
Display full name

### PR DESCRIPTION
**Pull request by kimbj95**
The setting `DEFAULT_SHOW_FULL_NAME` promises to use the user's full name everywhere it can be used.

Unfortunately the function `*user_model.User.ShortName()` currently uses the `.Name` instead - but this should also use the `.FullName()`.

Therefore we should make `*user_model.User.ShortName()` base its pre-shortened name on the `.FullName()` function.

### CODE REVIEW

File: models/db/name.go [hunk 1]
```diff
@@ -18,6 +18,9 @@ var (
 
 	// AlphaDashDotPattern characters prohibited in a user name (anything except A-Za-z0-9_.-)
 	AlphaDashDotPattern = regexp.MustCompile(`[^\w-\.]`)
+
+	// AlphaDashDotPattern characters prohibited in a user name (anything except A-Za-z0-9_.-)
+	AlphaDashDotPipePattern = regexp.MustCompile(`[^\w-\.\|]`)
```
Comment (a1012112796) [inline models/db/name.go]: why add this?

Comment (kimbj95) [reply models/db/name.go]: Mixed with other branch. Fixed. Thanks!

**Comment by zeripath** [comment]
_No description provided._

Please could you write something about what your PR is supposed to be doing and why you think this solves this problem.

I appreciate that this is a oneline PR but please TELL US MORE about what it is doing

**Review by zeripath — APPROVED** [review]

**Comment by kimbj95** [comment]
@zeripath I added the description. Sorry for not having the description.

**Review by wxiaoguang — APPROVED** [review]

**Comment by zeripath** [comment]
make lgtm work

**Pull request merged**